package bonus3;

import java.io.IOException;

public class Test {

    public static void main(String[] args) {
        String fileName = "animal_data.txt";
        Dog bobby = new Dog("Bobby", 3, "White", "Sherpherd");

        try {
            FilePart.saveToFile(bobby, fileName);
            System.out.println("Объект сохранен в файл: " + fileName);

            Dog restoredDog = (Dog) FilePart.readFromFile(fileName, true);
            System.out.println("Восстановленный объект из файла:");
            System.out.println(restoredDog);

        } catch (IOException e) {
            System.err.println("Ошибка при работе с файлом: " + e.getMessage());
        }
    }
}
